lib/Db/MessageMapper.php

```diff
@@ -178,15 +179,10 @@ public function findUidsForIds(Mailbox $mailbox, array $ids) {
 				$query->expr()->in('id', $query->createParameter('ids'), IQueryBuilder::PARAM_INT_ARRAY)
 			);
 
-		$chunks = array_chunk($ids, 1000);
-
-		$results = [];
-		foreach ($chunks as $chunk) {
+		return array_flat_map(function(array $chunk) use ($query) {
 			$query->setParameter('ids', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
-			$results[] = $this->findUids($query);
-		}
-
-		return array_merge(...$results);
+			return $this->findUids($query);
+		}, array_chunk($ids, 1000));
```

Comment:
move array_chunk also inside?

Reply:
inside the new function? Nah, that is too specific for this use case